File: atcoder/abc115/c.cpp
#include <bits/stdc++.h>
using namespace std;
#define rep(i,N) for(int i=0;i<int(N);++i)
#define all(a) (a).begin(),(a).end()

typedef long long ll;
typedef vector<int> vi;
typedef set<int> seti;
typedef vector<string> vs;

const int MOD = 1e9 + 7;
const int inf = 1e10;

int main() {
	int N,K;
	cin>>N>>K;
	vector<int> h(N);
	rep(i,N)cin>>h[i];
	sort(all(h));
	int ans = inf;
	rep(i,N){
		if(i+K-1<N)ans = min(ans,h[i+K-1]-h[i]);
	}
	cout<<ans<<endl;
}

File: atcoder/abc115/c.py
N, K = map(int,input().split( ))
H =[]
for i in range(N):
	h = int(input())
	H.append(h)
ansl = []
H.sort()

for i in range(N - K + 1):
	#HK = H[i:i+K]
	#ans = max(HK)-min(HK)
	ans = H[i + K - 1] - H[i]
	ansl.append(ans)

print(min(ansl))
